script/tool/lib/src/common/core.dart

```diff
@@ -53,17 +53,35 @@ bool isFlutterPackage(FileSystemEntity entity) {
   }
 }
 
+/// Prints `successMessage` in green.
+void printSuccess(String successMessage) {
+  print(Colorize(successMessage)..green());
+}
+
 /// Prints `errorMessage` in red.
 void printError(String errorMessage) {
-  final Colorize redError = Colorize(errorMessage)..red();
-  print(redError);
+  print(Colorize(errorMessage)..red());
 }
 
 /// Error thrown when a command needs to exit with a non-zero exit code.
+///
+/// While there is no specific definition of the meaning of different non-zero
+/// exit codes for this tool, commands should follow the general convention:
+///   1: The command ran correctly, but found errors.
+///   2: The command failed to run because the arguments were invalid.
+///  >2: The command failed to run correctly for some other reason. Ideally,
+///      each such failure should have a unique exit code within the context of
+///      that command.
 class ToolExit extends Error {
   /// Creates a tool exit with the given [exitCode].
   ToolExit(this.exitCode);
 
   /// The code that the process should exit with.
   final int exitCode;
 }
+
+/// A exit code for [ToolExit] for a successful run that found errors.
+const kExitCommandFoundErrors = 1;
+
+/// A exit code for [ToolExit] for a failure to run due to invalid arguments.
+const kExitInvalidArguments = 2;
```

Comment:
nit: I don't think the k prefix is correct Dart style: https://dart.dev/guides/language/effective-dart/style#prefer-using-lowercamelcase-for-constant-names  I know you started doing that in a different PR.  I'm just realizing now, sorry.  It's not a huge deal to me personally.